components/benefit_sponsors/app/controllers/benefit_sponsors/profiles/employers/employer_profiles_controller.rb

```diff
@@ -109,6 +109,8 @@ def export_census_employees
         end
 
         def show_invoice
+          authorize @employer_profile, :show?
```

Comment:
add these methods in policy class and refer `show?` in it.

Reply:
the `show?` method is already defined in the `EmployerProfilePolicy` and it appears to be functioning as intended.

Here's the current implementation for reference:

```
module BenefitSponsors
  class EmployerProfilePolicy < ::ApplicationPolicy

    def show?
      return false unless user.present?
      return true if user.has_hbx_staff_role? || is_broker_for_employer?(record) || is_general_agency_staff_for_employer?(record)
      is_staff_role_for_employer?
    end
  end
end
```

Reply:
no, what i meant is - you should define below in your policy

```
def show_invoice?
 show?
end

def download_invoice?
 show?
end

def estimate_cost?
 show?
end
```

Reply:
Actually, I noticed we've used authorize @employer_profile, :show? in the past, for one of the action bulk_employee_upload which was added by you. I'm happy to update my code to define separate policy methods for each action for better maintainability. Should I update bulk_employee_upload too?

Reply:
commits/code you're referring to are from 2018. above is the pattern we're following on security items from past 2-3 months.